### ISSUE TITLE
Topology doesn't reflect correctly across nodes

### ISSUE THREAD

**Issue by justinflipflops** [seaweedfs/seaweedfs#4113]
When settings up a multi master cluster with volume servers. If I use a command which utilizes hosts:

weed -v=3 master -ip=swfs00pp.mydomain.net -ip.bind=0.0.0.0 -port=9333 -mdir=./1 -peers=swfs00pp.mydomain.net:9333,swfs00pp.mydomain.net:9334,swfs00pp.mydomain.net:9335

and repeat that 3 times changing the port for each extra node.

then I added a volume server with:
weed -v=3 volume -dir=/opt/data/disk00 -ip=swfs00pp.mydomain.net -port=8080 -ip.bind=0.0.0.0 -mserver=swfs00pp.mydomain.net:9333,swfs00pp.mydomain.net:9334,swfs00pp.mydomain.net:9335

when I navigate to the master web interface, the topology only reflects the volumes on the primary master. The other servers will not show anything when I view them for volumes.

The hostname resolves to a public IP address assigned to the machine. If I change all occurrences of swfs00pp.mydomain.net to the IP address everything works correctly. 

Is this expected behavior? It works correctly using "localhost" as well. Just not a custom hostname/domain.

**Comment by chrislusf** [comment]
seems something wrong with hostname resolving.

the host name will be tried to resolve to ip address, and then used as part of the server identifier. you may want to use ip addresses directly.

**Comment by justinflipflops** [comment]
@chrislusf  thanks for the quick reply, I'll check further, but they are all using systemd-resolved with the addressing setup in /etc/hosts, so it never queries outside of themselves for the IP address and there is only one IP address assigned to the hostname. I thought it may have been because they are just dns records and not the FQDN of the servers themselves, so I even attempted with the true FQDN with the same results. 

I did get everything working via IP address, but now I'm experiencing extreme performance issues I did not have on the same equipment a few months back when testing various file systems before making a decision. 

Let's go ahead and close this, because it does make me feel there is another underlying issue that may be at play.

Again thanks for the prompt response @chrislusf
